### PR TITLE
Migrate from formation-react to component-library

### LINKED ISSUE #17254: Use new component-library package
## Issue Description

Update `vets-website` to use the new `component-library` package

---
## Tasks
- [ ] Update `package.json` in `vets-website` to use `component-library` instead of `formation-react`

## Acceptance Criteria
- [ ] `vets-website` uses `component-library`

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/17254

This is a simple find-and-replace of `formation-react` with `component-library` as we transition to that new repo. The command I ran to do this bulk update was
```sh
rg "formation-react" --files-with-matches | xargs -I {} sed -Ei 's/formation-react/component-library/g' {}
```

## Testing done
Testing in CI

## Screenshots
N/A

## Acceptance criteria
- [x] `formation-react` is no longer used in `vets-website`
- [x] `component-library` is used instead
- [ ] No breaking changes are introduced

### CODE REVIEW

**Comment by kelsonic** [comment]
Interesting, so does the `component-library` include more than just React components now? I just don't have context of why there's been a name change 🙂

**Comment by cvalarida** [comment]
Great question, @kelsonic! There were a few reason we did this:
1. We're moving away from the name `formation` for our design system
2. Having our component library in the `veteran-facing-services-tools` was causing a lot of headaches
   - Having Lerna manage the dependencies was a pain in the butt
   - Getting PRs merged took longer than it needed to because of the documentation site build in that same repo

We also decided to drop the `react` from the name, since we wanted this to be flexible enough to accommodate any technology changes we may decide on down the road—for example, if we wanted to migrate to web components which could be used with React _and_ in the static, Drupal-managed pages.

**Review by mdewey — APPROVED** [review]
LGTM.

**Review by ncksllvn — APPROVED** [review]

**Review by asg5704 — APPROVED** [review]
LGTM in both the `vets-website` & `vsp-test-user-dashboard` repos

**Review by micahchiang — APPROVED** [review]
472 🔥

**Review by stephenBarrs — APPROVED** [review]
LGTM!

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -325,6 +325,7 @@
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-dropzone": "^4.2.5",
+    "react-focus-lock": "^2.5.0",
```
Comment (bkjohnson) [inline package.json:328]: What is this for?

**Review by Mottie — APPROVED** [review]
ghe-emoji

**Review by jbalboni — APPROVED** [review]

**Review by LeKeve — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (cvalarida) [reply package.json:328]: The `MegaMenu` [uses it](https://github.com/department-of-veterans-affairs/vets-website/blob/bb3e023828eac88dfc686fad59b9bd0ffc3df166/src/platform/site-wide/mega-menu/components/SubMenu.jsx#L3). It's unclear to me why this was in `node_modules` before but isn't after this change, though. At any rate, it's a direct dependency of `vets-website`, so it should be in the dependency list here now. :man_shrugging:

**Review by bkjohnson — APPROVED** [review]

**Review by mpelzsherman — APPROVED** [review]

**Review by U-DON — APPROVED** [review]

**Review by Clark-W-Griswold — APPROVED** [review]
👍

**Review by erikphansen — APPROVED** [review]

**Review by bradl3yC — APPROVED** [review]

**Review by LinnJS — APPROVED** [review]

**Comment by mchelen-gov** [comment]
@cvalarida
> Great question, @kelsonic! There were a few reason we did this:
> 
> 1. We're moving away from the name `formation` for our design system
> 2. Having our component library in the `veteran-facing-services-tools` was causing a lot of headaches
>    
>    * Having Lerna manage the dependencies was a pain in the butt
>    * Getting PRs merged took longer than it needed to because of the documentation site build in that same repo
> 
> We also decided to drop the `react` from the name, since we wanted this to be flexible enough to accommodate any technology changes we may decide on down the road—for example, if we wanted to migrate to web components which could be used with React _and_ in the static, Drupal-managed pages.

Yes! Been hoping to see this. New repo looks like: https://github.com/department-of-veterans-affairs/component-library

**Review by mchelen-gov — APPROVED** [review]

**Pull request merged**